### PR TITLE
Frontend feature messages tab

### PR DESCRIPTION
**Pull request by amdwit**
An empty messages page with just the title and tabbar

### CODE REVIEW

File: app/frontend/src/AppRoutes.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Switch, Route, RouteProps, Redirect } from "react-router-dom";
 import Login from "./features/auth/login/Login";
 import ProfilePage from "./features/profile/ProfilePage";
 import Home from "./features/Home";
-import Messages from "./features/Messages";
+import Messages from "./features/messages/index";
```
Comment (lucaslcode) [inline app/frontend/src/AppRoutes.tsx:6]: Why this change?

File: app/frontend/src/features/messages/Messages.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <>
+      <PageTitle>Messages</PageTitle>
+      <TabContext value={value}>
+        <TabBar value={value} setValue={setValue} labels={labels} />
+        <TabPanel value={"TAB_ALL"}>ALL</TabPanel>
+        <TabPanel value={"TAB_GROUPCHATS"}>GROUPCHATS</TabPanel>
+        <TabPanel value={"TAB_HOSTING"}>HOSTING</TabPanel>
+        <TabPanel value={"TAB_SURFING"}>SURFING</TabPanel>
+        <TabPanel value={"TAB_MEET"}>MEET</TabPanel>
+        <TabPanel value={"TAB_ARCHIVED"}>ARCHIVED</TabPanel>
```
Comment (lucaslcode) [inline app/frontend/src/features/messages/Messages.tsx:19]: ~~Group chats with a space. Also I think this should be sentence case and use css transforms for all caps (or just sentence case for now).~~ Nevermind these are just the placeholders

File: app/frontend/src/features/messages/Messages.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import TabBar from "./TabBar";
+import { labels } from "./constants";
+
+export default function Messages() {
+  const [value, setValue] = React.useState<keyof typeof labels>("TAB_ALL");
+  return (
+    <>
+      <PageTitle>Messages</PageTitle>
+      <TabContext value={value}>
+        <TabBar value={value} setValue={setValue} labels={labels} />
```
Comment (lucaslcode) [inline app/frontend/src/features/messages/Messages.tsx:13]: This needs `scrollButton="auto"` or something like that, I can't scroll when doing mobile preview on Edge.

File: app/frontend/src/features/messages/TabBar.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  labels: T;
+}
+
+export default function TabBar<T extends Record<string, string>>({ value, setValue, labels }: TabBarProps<T>) {
+  const handleChange = (event: any, newValue: keyof T) => {
+    setValue(newValue);
+  };
+  const classes = useStyles();
+  return (
+    <TabList
```
Comment (lucaslcode) [inline app/frontend/src/features/messages/TabBar.tsx]: What is the difference between TabList and Tabs

**Review by lucaslcode — DISMISSED** [review]
lgtm!

File: app/frontend/src/AppRoutes.tsx (hunk 1 above)
Comment (amdwit) [reply app/frontend/src/AppRoutes.tsx:6]: Messages is a file, messages is a directory.

File: app/frontend/src/features/messages/Messages.tsx (hunk 2 above)
Comment (amdwit) [reply app/frontend/src/features/messages/Messages.tsx:19]: But I agree with the idea to not use layout in text.

File: app/frontend/src/features/messages/TabBar.tsx (hunk 4 above)
Comment (amdwit) [reply app/frontend/src/features/messages/TabBar.tsx]: I do not know exactly. I see the children props is of different type and on the demo page TabList is related to accessibility.

File: app/frontend/src/features/messages/Messages.tsx (hunk 3 above)
Comment (amdwit) [reply app/frontend/src/features/messages/Messages.tsx:13]: Fixed

**Review by lucaslcode — APPROVED** [review]
